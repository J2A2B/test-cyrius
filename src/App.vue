<template>
  <div>
    <button id="press-to-display" @click="displayMessage">
      {{ buttonText }}
    </button>
    <QuizTest v-if="!data.isClosed" :title-quiz="WORDINGPAGE.titleQuiz" />
  </div>
</template>

<script lang="ts" setup>
import { reactive, computed } from "vue";
import QuizTest from "./components/QuizTest.vue";

interface WordingPage {
  titleQuiz: string;
  closeQuiz: string;
  openQuiz: string;
}

const WORDINGPAGE: WordingPage = {
  titleQuiz: "Mon super Quiz",
  closeQuiz: "Fermer le Quiz",
  openQuiz: "Lancer le Quiz",
};
const data = reactive<{ isClosed: boolean }>({
  isClosed: true,
});

const displayMessage = () => {
  data.isClosed = !data.isClosed;
};
const buttonText = computed<string>(() =>
  data.isClosed ? WORDINGPAGE.openQuiz : WORDINGPAGE.closeQuiz
);
</script>

<style>
#app {
  font-family: Avenir, Helvetica, Arial, sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  text-align: center;
  color: #2c3e50;
  margin-top: 60px;
}
button {
  margin: 10px;
  padding: 10px;
  background-color: #ff3660;
  color: white;
  border: none;
  border-radius: 3px;
  cursor: pointer;
  font-size: 18px;
}
</style>
