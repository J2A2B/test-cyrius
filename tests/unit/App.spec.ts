import { shallowMount, mount } from "@vue/test-utils";
import App from "@/App.vue";

describe("App.vue", () => {
  describe("Template", () => {
    it("should match snapshot", () => {
      const wrapper = shallowMount(App);

      expect(wrapper.element).toMatchSnapshot();
    });
    describe("When the Quiz is open", () => {
      it("should match snapshot", async () => {
        const wrapper = shallowMount(App);

        wrapper.vm.data.isClosed = false;
        await wrapper.vm.$nextTick();

        expect(wrapper.element).toMatchSnapshot();
      });
    });
  });
});
describe("Data", () => {
  it("should set the default Data values", async () => {
    const wrapper = mount(App);

    await expect(wrapper.vm.data.isClosed).toEqual(true);
    await expect(wrapper.vm.buttonText).toEqual("Lancer le Quiz");
  });
});
describe("On click on the button", () => {
  it("should set the data value to true", async () => {
    const wrapper = mount(App);
    await wrapper.find("#press-to-display").trigger("click");

    await expect(wrapper.vm.data.isClosed).toEqual(false);
  });
  it("should set the button text to Fermer le Quiz", async () => {
    const wrapper = mount(App);
    await wrapper.find("#press-to-display").trigger("click");

    await expect(wrapper.vm.buttonText).toEqual("Fermer le Quiz");
  });
});
