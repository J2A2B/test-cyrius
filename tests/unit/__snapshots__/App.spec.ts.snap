// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue Template When the Quiz is open should match snapshot 1`] = `
<div>
  <button
    id="press-to-display"
  >
    Fermer le Quiz
  </button>
  <quiz-test-stub
    titlequiz="Mon super Quiz"
  />
</div>
`;

exports[`App.vue Template should match snapshot 1`] = `
<div>
  <button
    id="press-to-display"
  >
    Lancer le Quiz
  </button>
  <!--v-if-->
</div>
`;

exports[`App.vue Template should match snapshot2 1`] = `
<div>
  <button
    id="press-to-display"
  >
    Lancer le Quiz
  </button>
  <!--v-if-->
</div>
`;
